### ISSUE TITLE
Check if Data Type implementes Stringable interfacr for Primary keys

### ISSUE THREAD

**Issue by michnovka** [doctrine/dbal#5313]
### Feature Request

I came across [this issue](https://stackoverflow.com/questions/15080573/doctrine-2-orm-datetime-field-in-identifier/71401564#71401564) recently

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | no

#### Summary

I think that if field has defined type in PHP Entity, then Doctrine should check whether this type implements Stringable interface (has __toString() method). If not, an error should be thrown, as UnitOfWork cannot keep track of changes and whole ORM is thus broken.

**Comment by derrabus** [comment]
I wonder if we should rather leverage the type system here. After all, it's the database representation of the ID and the type system could get us that representation.

**Comment by michnovka** [comment]
@derrabus yes, I agree. We could add `Type::convertToPHPString(): string` method or `convertToPHPStringable(): \Stringable`

**Comment by michnovka** [comment]
Since the above would work only for PHP8, we can do instead

```php

    public function convertToPHPString($value, AbstractPlatform $platform): string
    {
        $value = self::convertToPHPValue($value, $platform);

        if (is_object($value) && false === method_exists($value, '__toString')) {
            throw new \LogicException("PHP value must be convertible to string");
        }

        return (string) $value;
    }
```
